### PR TITLE
fix(meta): expired RiseCtl worker causes meta panics

### LINKED ISSUE #5194: bug: notification service on meta enters unreachable code for risectl
When a `risectl` worker expired, meta service will enter unreachable code when `delete_sender`. Maybe introduced by #5132.

Note that not all workers register the notification service.

```
thread 'risingwave-main' panicked at 'internal error: entered unreachable code', src/meta/src/manager/notification.rs:154:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:48:5
   3: risingwave_meta::manager::notification::NotificationManager::delete_sender::{{closure}}
             at ./src/meta/src/manager/notification.rs:154:18
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   5: risingwave_meta::manager::cluster::ClusterManager<S>::start_heartbeat_checker::{{closure}}::{{closure}}
             at ./src/meta/src/manager/cluster.rs:282:84
```

https://github.com/risingwavelabs/risingwave/blob/3e0d4d2873c47c1bd141ccd90d286c75fff06e6f/src/meta/src/manager/notification.rs#L147-L156

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- As title.
- Use any command by risedev ctl. After 60s without any heartbeat, meta will panic.

Log:
```
thread 'risingwave-main' panicked at 'internal error: entered unreachable code', src/meta/src/manager/notification.rs:155:18
```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
close #5194

### CODE REVIEW

File: src/meta/src/manager/notification.rs [hunk 1]
```diff
@@ -151,6 +151,7 @@ impl NotificationManager {
             WorkerType::Frontend => core_guard.compute_senders.remove(&worker_key),
             WorkerType::ComputeNode => core_guard.frontend_senders.remove(&worker_key),
             WorkerType::Compactor => core_guard.compactor_senders.remove(&worker_key),
+            WorkerType::RiseCtl => None,
             _ => unreachable!(),
```
Comment (BugenZhao) [inline src/meta/src/manager/notification.rs:155]: BTW, what's `WorkerType::Generic` for? cc @yezizp2012 @Li0k

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Comment by HuaHuaY** [comment]
I think it should be modified on the heartbeat side, not on the notification manager. This problem is due to cluster manager executing `delete_sender` for each worker_node.

File: src/meta/src/manager/notification.rs (hunk 1 above)
Comment (yezizp2012) [reply src/meta/src/manager/notification.rs:155]: `WorkerType::Generic` is a generic worker type introduced by @Li0k in order to send notification to all nodes. But it seems like not needed anymore?

**Review by BugenZhao — APPROVED** [review]

**Review by Li0k — APPROVED** [review]

**Pull request merged**